File: ChipmeoPOS/src/lib/types/employee.ts
export interface Employee {
	id: number;
	fullName: string;
	username: string;
	email?: string;
	phone?: string;
	avatarUrl?: string | null;
	roleId: number;
	roleName?: string;
	isActive: boolean;
	createdAt?: string;
	updatedAt?: string;
}

export interface EmployeeCreateDto {
	fullName: string;
	username: string;
	password: string;
	email?: string;
	phone?: string;
	avatarUrl?: string | null;
	roleId: number;
	isActive: boolean;
}

export interface EmployeeUpdateDto {
	fullName: string;
	username: string;
	email?: string;
	phone?: string;
	avatarUrl?: string | null;
	roleId: number;
	isActive: boolean;
}

File: foodstore-api/FoodstoreApi.Core/Entities/Employee.cs
namespace FoodstoreApi.Core.Entities;

public partial class Employee
{
    public int Id { get; set; }

    public string FullName { get; set; } = null!;

    public string Username { get; set; } = null!;

    public string PasswordHash { get; set; } = null!;

    public string? Email { get; set; }

    public string? Phone { get; set; }

    public string? AvatarUrl { get; set; }

    public int RoleId { get; set; }

    public bool? IsActive { get; set; }

    public DateTime? LastLogin { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<Order> OrderEmployees { get; set; } = new List<Order>();

    public virtual ICollection<OrderStatusHistory> OrderStatusHistories { get; set; } = new List<OrderStatusHistory>();

    public virtual ICollection<Order> OrderUpdatedByNavigations { get; set; } = new List<Order>();

    public virtual ICollection<BlogPost> BlogPosts { get; set; } = new List<BlogPost>();

    public virtual Role Role { get; set; } = null!;
}




